app/controllers/pages/secondary_skip_controller.rb

```diff
@@ -54,6 +56,20 @@ def ensure_branch_routing_feature_enabled
     raise ActionController::RoutingError, "branch_routing feature not enabled" unless Settings.features.branch_routing
   end
 
+  def ensure_page_has_skip_condition
+    unless page.conditions.any? { |c| c.answer_value.present? }
+      redirect_to form_pages_path(current_form.id)
+    end
+  end
+
+  def ensure_secondary_skip_exists
+    redirect_to show_routes_path(form_id: current_form.id, page_id: page.id) if secondary_skip_condition.blank?
+  end
+
+  def ensure_secondary_skip_blank
+    redirect_to show_routes_path(form_id: current_form.id, page_id: page.id) if secondary_skip_condition.present?
+  end
```

Comment:
I think rather than doing a redirect here we should just return a 404, or even raise an exception; we shouldn't be directing users to these pages in these scenarios right?

Reply:
hmm, thats a good point. I was thinking the only time this would show up would be when a user had multiple tabs open. In that case a link might still point to new and I figured redirecting them back to the routes page would make sense rather than showing an error.

That might pretty confusing in use though so perhaps showing an error is better.

I'm going to merge this for now but I'm happy to change it.

Reply:
Ooh, that's a good point about multiple tabs, I hadn't thought about that. Something to think more about then.